### PR TITLE
Veoss 1177 | Align registration submission model to ETMP

### PR DESCRIPTION
**Pull request by warrenskiwork1979**

### CODE REVIEW

File: app/models/etmp/EtmpSchemeDetails.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+package models.etmp
+
+import play.api.libs.json.{Json, OFormat}
+
+import java.time.format.DateTimeFormatter
+
+case class EtmpSchemeDetails(commencementDate: String,
+                             firstSaleDate: Option[String],
+                             euRegistrationDetails: Option[Seq[EtmpEuRegistrationDetails]],
+                             previousEURegistrationDetails: Option[Seq[EtmpPreviousEURegistrationDetails]],
```
Comment (DevAndyroo) [inline app/models/etmp/EtmpSchemeDetails.scala:26]: For both the sequences, we shouldn't make them optional, I know they're optional in the schema, but the model should allow empty sequence

File: app/models/etmp/EtmpRegistrationRequest.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                             vrn: Vrn,
+                             tradingNames: Seq[EtmpTradingNames],
+                             schemeDetails: EtmpSchemeDetails,
+                             bankDetails: BankDetails
+                           ) {
+}
+
+object EtmpRegistrationRequest {
+
+  def fromRegistrationRequest(registration: RegistrationRequest): EtmpRegistrationRequest = {
```
Comment (DevAndyroo) [inline app/models/etmp/EtmpRegistrationRequest.scala:34]: can we add some specific unit tests for this? Ensuring the converted model matches what we expect

**Review by DevAndyroo — CHANGES_REQUESTED** [review]

File: app/models/etmp/EtmpSchemeDetails.scala (hunk 1 above)
Comment (warrenskiwork1979) [reply app/models/etmp/EtmpSchemeDetails.scala:26]: Changed both from option[Seq] to Seq

File: app/models/etmp/EtmpRegistrationRequest.scala (hunk 2 above)
Comment (warrenskiwork1979) [reply app/models/etmp/EtmpRegistrationRequest.scala:34]: Added unit test for model conversion.

**Review by DevAndyroo — APPROVED** [review]

**Pull request merged**